### PR TITLE
ErrorAnalyzer: change names/values for error tags

### PR DESCRIPTION
**Pull request by johnbley**
- Don't emit error.msg, error.type or error.stack tags
- Do emit sfx.error.{message, object, kind, stack} tags, with values appropriate to the OpenTracing semantic conventions at https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table
- Fix up tests accordingly

### CODE REVIEW

File: dd-java-agent/instrumentation/jsp-2.3/src/test/groovy/JSPInstrumentationBasicTests.groovy [hunk 1]
```diff
@@ -342,13 +343,14 @@ class JSPInstrumentationBasicTests extends AgentTestRunner {
             "servlet.context" "/$jspWebappContext"
             "jsp.requestURL" reqUrl
             "error" true
-            "error.type" { String tagExceptionType ->
+            "sfx.error.object" { String tagExceptionType ->
```
Comment (rmfitzpatrick) [inline dd-java-agent/instrumentation/jsp-2.3/src/test/groovy/JSPInstrumentationBasicTests.groovy]: Mind adding the modified header to this file?

File: dd-trace-ot/src/main/java/datadog/opentracing/DDSpan.java [hunk 2]
```diff
@@ -152,8 +152,10 @@ public MutableSpan getLocalRootSpan() {
   public void setErrorMeta(final Throwable error) {
     setError(true);
 
+    final String type = error.getClass().getName();
     setTag(DDTags.ERROR_MSG, error.getMessage());
     setTag(DDTags.ERROR_TYPE, error.getClass().getName());
```
Comment (rmfitzpatrick) [inline dd-trace-ot/src/main/java/datadog/opentracing/DDSpan.java]: Did you mean to use `type` here?

**Comment by rmfitzpatrick** [comment]
We'll need to clearly detail this in the next release logs.

**Comment by owais** [comment]
Do we continue to use `error: true` boolean tag or migrate it to `sfx.error = true` as well? IIRC, zipkin sets the `error` tag to a human readable error message. As a result it might not always be a boolean across different languages. @rmfitzpatrick @johnbley @flands

**Comment by owais** [comment]
We can still set `error: <value>` in the final formatter/translation layer when converting to zipkin.

**Comment by owais** [comment]
Actually, sticking to `error` is fine. I'll update all cases where we set `error` to strings in the node package.

**Comment by rmfitzpatrick** [comment]
> Do we continue to use `error: true` boolean tag or migrate it to `sfx.error = true` as well? IIRC, zipkin sets the `error` tag to a human readable error message. As a result it might not always be a boolean across different languages. @rmfitzpatrick @johnbley @flands

Our convention thus far has been OpenTracing's where `error` is a bool.

**Comment by owais** [comment]
@rmfitzpatrick Node package sets error to `"true"` (string) instead of bool. Should this be updated/fixed?

**Pull request merged**